From 14dca6a04af1828917ab1602ff8b1d225411bf7b Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Wed, 16 Apr 2025 00:29:41 +0200
Subject: [PATCH 29/29] libcxx: default to type 2

// This implementation of type_info does not assume there is always a unique
// copy of the RTTI for a given type inside a program. For various reasons
// the linker may have failed to merge every copy of a types RTTI
// (For example: -Bsymbolic or llvm.org/PR37398). Under this assumption, two
// type_infos are equal if their addresses are equal or if a deep string
// comparison is equal.

this is mostly for compat with libstdc++, which some projects
like kde* rely on:

https://bugs.kde.org/show_bug.cgi?id=479679
---
 libcxx/include/typeinfo | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libcxx/include/typeinfo b/libcxx/include/typeinfo
index 799c6ebd5..12572e646 100644
--- a/libcxx/include/typeinfo
+++ b/libcxx/include/typeinfo
@@ -182,7 +182,7 @@ public:
 
 // On all other platforms, assume the Itanium C++ ABI and use the Unique implementation.
 #        else
-#          define _LIBCPP_TYPEINFO_COMPARISON_IMPLEMENTATION 1
+#          define _LIBCPP_TYPEINFO_COMPARISON_IMPLEMENTATION 2
 #        endif
 #      endif
 
-- 
2.49.0

